### PR TITLE
added utils::getCascadeBoundingBox()

### PR DESCRIPTION
**Pull request by SunLightJuly**
This boundingBox will calculate all children's boundingBox every time

### CODE REVIEW

File: cocos/2d/CCNode.h [hunk 1]
```diff
@@ -1079,6 +1079,11 @@ class CC_DLL Node : public Ref
     /** @deprecated Use getBoundingBox instead */
     CC_DEPRECATED_ATTRIBUTE inline virtual Rect boundingBox() const { return getBoundingBox(); }
 
+    /**
+     * This boundingBox will calculate all children's boundingBox every time
+     */
+    virtual Rect getCascadeBoundingBox(void);
```
Comment (minggo) [inline cocos/2d/CCNode.h]: I think it is better to be a helper function into `utils`.

File: cocos/2d/CCNode.cpp [hunk 2]
```diff
@@ -803,6 +803,49 @@ Rect Node::getBoundingBox() const
     return RectApplyAffineTransform(rect, getNodeToParentAffineTransform());
 }
 
+Rect Node::getCascadeBoundingBox(void)
+{
+    Rect cbb;
+
+    // check all childrens bounding box, get maximize box
+    Node* child = NULL;
```
Comment (minggo) [inline cocos/2d/CCNode.cpp]: NULL -> nullptr

Comment (SunLightJuly) [inline cocos/2d/CCNode.cpp]: fixed

File: cocos/2d/CCNode.h (hunk 1 above)
Comment (SunLightJuly) [inline cocos/2d/CCNode.h]: moved

**Pull request merged**